docs/user_guide/hdl_coding_guideline.rst

```diff
@@ -0,0 +1,960 @@
+   --
+   -- Redistribution and use of source or resulting binaries, with or without modification
+   -- of this file, are permitted under one of the following two license terms:
+   --
+   --   1. The GNU General Public License version 2 as published by the
+   --      Free Software Foundation, which can be found in the top level directory
+   --      of this repository (LICENSE_GPL2), and also online at:
+   --      <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
+   --
+   -- OR
+   --
+   --   2. An ADI specific BSD license, which can be found in the top level directory
+   --      of this repository (LICENSE_ADIBSD), and also on-line at:
+   --      https://github.com/analogdevicesinc/hdl/blob/master/LICENSE_ADIBSD
+   --      This will allow to generate bit files and not release the source code,
+   --      as long as it attaches to an ADI device.
+   --
+   -- ***************************************************************************
+   -- ***************************************************************************
+
+   entity prescaler is
+     Port (
+       core_32m_clk      : in  std_logic,    -- 32 MHz clock
+       system_clk        : in  std_logic,    -- system clock
+       scan_mode_test    : in  std_logic,    -- scan mode clock
+       reset_n           : in  std_logic,    -- active low hard reset, synch
+       -- w/ system_clock
+       div16_clk         : out std_logic,    -- input clock divided by 16
+       div16_clk_n       : out std_logic     -- input clock divided by 16
+        -- and inverted
+     );
+   end prescaler;
+
+   architecture Behavioral of  prescaler is
+
+   -- Components Declarations
+
+   -- Local Types Declarations
+
+   --  Constants Declarations
+
+   -- Signals Declarations
+     signal count        : std_logic_vector(3 downto 0); -- 4-bit counter to
+     -- make clock divider
+     signal count_ns     : std_logic_vector(3 downto 0); -- clock divider next
+     -- state input
+
+   -- Module Implementation
+   begin
+
+     -- This process updates the internal counter
+     process(core_32m_clk)
+     begin
+       if (rising_edge(core_32m_clk)) then
+         if (reset_n = '0') then
+           -- reset counter
+           count <= "0000";
+         else
+           -- update counter
+           count <= count + "0001";
+         end if;
+       end if;
+     end process;
+
+     -- This process updates the output clock signals
+     process(scan_mode_test, system_clk, count)
+     begin
+       if (scan_mode_test = '0') then
+         -- normal operation clock assign
+         div16_clk <= count(3);
+         div16_clk_n <= not count(3);
+       else
+         -- scan mode clock assign
+         div16_clk <= system_clk;
+         div16_clk_n <= system_clk;
+       end if;
+     end process;
+
+   end Behavioral;
+
+4. References
+-------------------------------------------------------------------------------
+
+`1 <http://www.xilinx.com/support/documentation/white_papers/wp231.pdf>`__
```

Comment:
These are inconsistent.
Perhaps make all follow this format:

```
[5] Author, "Title", Date, ..., available at:
`shortened url <full url>`_
```

Reply:
I edited it, let me know if it looks ok like this